CMakeLists.txt

```diff
@@ -1,189 +1,307 @@
+    # Standard definitions
+    # -DWIN32 -D_WIN32 -D_WINDOWS           = Windows platform
+    # -DNOMINMAX                            = Use the standard's templated min/max
+    add_definitions(-DWIN32 -D_WIN32 -D_WINDOWS -DWINDOWS -DNOMINMAX)
+    if(X64)
+        add_definitions(-DWIN64)
     endif()
-else()
-    set(BUILD_C_FLAGS "-fomit-frame-pointer -Wall -fsigned-char")
-    set(BUILD_CXX_FLAGS "-fno-rtti")
-    set(BUILD_DEBUG_FLAGS   "${BUILD_FLAGS} -O0 -g -D_DEBUG -DBUILD_DEBUG")
-    set(BUILD_RELEASE_FLAGS "${BUILD_FLAGS} -O3 -DNDEBUG -DBUILD_NO_DEBUG")
-endif()
 
-set(BUILD_C_DEBUG_FLAGS "${BUILD_C_FLAGS} ${BUILD_DEBUG_FLAGS}")
-set(BUILD_CXX_DEBUG_FLAGS "${BUILD_CXX_FLAGS} ${BUILD_DEBUG_FLAGS}")
+    # Configure use of the sandbox.
+    set(USE_SANDBOX OFF)
+    if(MSVC_VERSION EQUAL 1800)
+        # The I_sandbox.lib static library is currently built with VS2013. It will
+        # not link successfully with other VS versions.
+        set(USE_SANDBOX ON)
+        option(USE_SANDBOX "Enable or disable use of the sandbox. (VS 2013 or later required)" ON)
+    endif()
 
-set(BUILD_C_RELEASE_FLAGS "${BUILD_C_FLAGS} ${BUILD_RELEASE_FLAGS}")
-set(BUILD_CXX_RELEASE_FLAGS "${BUILD_CXX_FLAGS} ${BUILD_RELEASE_FLAGS}")
+    if(USE_SANDBOX)
+        # Definition required by CEF_sandbox.lib.
+        add_definitions(-DPSAPI_VERSION=1)
+        # Definition used by apps to test if the sandbox is enabled.
+        add_definitions(-DCEF_USE_SANDBOX)
+    endif()
 
-set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} ${BUILD_C_DEBUG_FLAGS}")
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${BUILD_CXX_DEBUG_FLAGS}")
-message(STATUS "Set debug mode flags (${BUILD_DEBUG_FLAGS}) (Current mode: ${CMAKE_BUILD_TYPE}, compiler: ${CMAKE_CXX_COMPILER_ID})")
+    # Rename project_arch from x86_64 to 64 and x86 to 32 on windows:
+    if(X64)
+        set(PROJECT_ARCH "64")
+    else()
+        set(PROJECT_ARCH "32")
+    endif()
 
-set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} ${BUILD_C_RELEASE_FLAGS}")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${BUILD_CXX_RELEASE_FLAGS}")
-message(STATUS "Set release mode flags (${BUILD_RELEASE_FLAGS}) (Current mode: ${CMAKE_BUILD_TYPE}, compiler: ${CMAKE_CXX_COMPILER_ID})")
+    # Define where to move the binary by install
+    set(EXE_SUB_DIR "bin/windows/${PROJECT_ARCH}")
 
-# Temp file directories
-set(BUILD_TEMP_DIR         ${CMAKE_BINARY_DIR}/tmp)
-set(BUILD_TEMP_INCLUDE_DIR ${BUILD_TEMP_DIR}/include)
-file(MAKE_DIRECTORY        ${BUILD_TEMP_DIR})
-file(MAKE_DIRECTORY        ${BUILD_TEMP_INCLUDE_DIR})
+    # Submodule providing windows-specific build stuff (e.g. libs)
+    set(PLATFORM_FILE "${SOURCE_DIR}/platform_windows")
+endif()
 
-# Detect date, TODO: use time command if available
-set(DATE_FILE ${BUILD_TEMP_DIR}/date CACHE INTERNAL "Temp date file path.")
+# Merge compiler/linker flags.
+set(CMAKE_C_FLAGS                     "${I_COMPILER_FLAGS} ${I_C_COMPILER_FLAGS}")
```

Comment:
Actually that was one of the main stuff i wanted to get rid of: otherwise i had like no control over the whole system.
This way the effects are more predictable imo.